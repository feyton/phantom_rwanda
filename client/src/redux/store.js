import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { busMovementReducer } from './reducers/busMovementReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { simulateReducer } from './reducers/simulateReducer.js';
import { socketReducer } from './reducers/socketReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer,
    socket: socketReducer,
    simulate: simulateReducer,
    movement: busMovementReducer
  }
});
