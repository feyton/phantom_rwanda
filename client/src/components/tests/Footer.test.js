import { mount } from 'enzyme';
import { default as React } from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import Footer from '../Footer.js';

const wrapper = mount(
  <Provider store={store}>
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  </Provider>
);

describe('Footer', () => {
  it('should render Footer', () => {
    const footer = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Footer />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();
    expect(footer).toMatchSnapshot();
  });
  it('contains one child elements element', () => {
    const value = wrapper.children().length;
    expect(value).toEqual(1);
  });
});
