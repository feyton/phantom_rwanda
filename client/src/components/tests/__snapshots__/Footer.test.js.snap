// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render Footer 1`] = `
<footer
  className="flex bg-primary z-20 xl:justify-center md:text-xl text-background font-sans font-sm py-8 lg:py-2 xl:py-2 mt-auto"
>
  <div
    className="flex flex-col lg:flex-row xl:flex-row px-12 text-center items-center"
  >
    <img
      alt="Codebandits"
      className="w-24 ml-2 lg:ml-8 lg:mt-4 xl:mt-2"
      src="codebandits.png"
    />
    <h3
      className="lg:ml-8 xl:ml-20 mt-6 xl:mt-2"
    >
      Â©
      copyrights
       Codebandits 2022
    </h3>
  </div>
  <div
    className="flex flex-col md:text-xl lg:flex-row xl:flex-row lg:mt-7 xl:mt-3 mr-16 md:mx-20 xl:ml-20 ml-auto md:ml-auto"
  >
    <a
      className="xl:ml-0 lg:ml-0 mt-2 text-md mb-4 hover:font-bold"
      href="/"
      onClick={[Function]}
    >
      Support
    </a>
    <a
      className="xl:ml-28 lg:ml-12 mt-2 text-md mb-4 hover:font-bold"
      href="/"
      onClick={[Function]}
    >
      terms
    </a>
    <a
      className="xl:ml-28 lg:ml-12 mt-2 text-md hover:font-bold"
      href="/login"
      onClick={[Function]}
    >
      administration
    </a>
  </div>
</footer>
`;
